src/main.rs

```diff
@@ -57,19 +45,52 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // Create an InputManager instance
     let mut input_manager = Manager::new(connection.clone());
 
-    // Start the input manager and listen on DBus
-    tokio::spawn(async move {
-        log::debug!("Starting input manager thread");
-        if let Err(e) = input_manager.run().await {
-            log::error!("Error running input manager: {:?}", e);
+    let (ctrl_c_result, input_man_result, request_name_result) = tokio::join!(
```

Comment:
I haven't ever experienced an issue with ctrl+c. When does this happen?

Reply:
The reference to the spawned thread is always lost in the original code as the tokio::spawn result was not even captured. I don't really like losing track of spawned tasks inside functions otherwise the need to debug with tokio console will intensify at the point of being a must to understand how things works together (even in the sense of sequential vs parallel)